### PR TITLE
Update to work with python 3

### PR DESCRIPTION
**Pull request by pganssle**
Changes:

* Updated to now support Python 2.7, 3.4 and 3.5
* Removed dependency on `simplestats`
* Added dependency on `six`, also for versions < 3.4, added dependency on `enum34`.
* **API Change**: Removed `enum` package (behavior superceded by addition of `enum` in Python 3, plus the backport)
* Increased code coverage (didn't get to 100%, but I went line-by-line on the missing lines to see any obvious compatibility issues).
* Made changes to `alternations`: All functions now work with both hiragana and katakana
* Made changes to `kana_table`:
    * **API Change**: `is_voiced` is implemented in a more general way, and now supports both *hiragana*, *katakana*, as well as anything that uses Japanese voicing diacritics (e.g. fake characters like ま゙.
   * Added an additional method `is_semivoiced`, which detects only semi-voiced characters like ぱ.
* Updated version number to 1.7.0 -  this can be version 2.0 because the removal of the `enum` package is a breaking API change, but I get the impression that was not really supposed to be part of the public interface.

I also would like to make some changes to how `cjkdata` is pulled in, but I figured that that is logically distinct from these changes, so it's probably best to do it in multiple stages. This is the "non-API breaking" portion of the PR that provides the same API (minus enum) but with Python 3 support. I think the `cjkdata` changes and possibly some others I've been mulling might be API-breaking.

### CODE REVIEW

File: cjktools/alternations.py [hunk 1]
```diff
@@ -106,69 +121,88 @@ def _create_voicing_map():
...
+
+            reverse_operation = script_converters[char_type]
+            segment = scripts.to_hiragana(segment)
+        else:
+            continue
+
+        for m in _long_finder.finditer(segment):
+            i = m.start() 
+            vowel = table.to_vowel_line(segment[i-1])
+            print(segment)
```
Comment (pganssle) [inline cjktools/alternations.py]: Oops. This is debugging detritus.

**Comment by larsyencken** [comment]
Paul, this is quite an impressive modernisation effort, basically touching every single file in the project, and a clean and well described pull request. I'm most happy to merge it.

I'll tag and release this as 1.7.0 as you suggest.

It seems to me that if you're going to be using dictionaries, you might be interested in JMdict's richer XML format rather than the old line & regex Edict style format that I used to parse. Check it out, see if it's useful.

**Comment by larsyencken** [comment]
It's up on PyPI now: https://pypi.python.org/pypi/cjktools

**Comment by pganssle** [comment]
@larsyencken Cool, thanks. FYI, the PyPI "page" link goes to bitbucket. I didn't realize that the bitbucket link wasn't a mirror of this repo so I didn't update it in `setup.py`. You can change that directly on PyPI even without a new release, IIRC.

**Pull request merged**